odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1571,7 +1571,7 @@ def _validate_tag_field(self, node, name_manager, node_info):
                         # most (~95%) elements are 1/True/0/False
                         res = str2bool(val)
                     except ValueError:
-                        res = safe_eval.safe_eval(val, {'context': self._context})
+                        res = bool(safe_eval.safe_eval(val, {'context': self._context}))
                     if res not in (1, 0, True, False, None):
```

Comment:
This check seems to be useless, as well as the test testing it

Reply:
why would the check be useless ?

This is validation of a view and enforcement of a spec
My first attempt crashed because of a test
- res at that point is an arbitrary string (not "1" or "True" etc....)
- res hence can be [[field, =, name]], or "somestring" (if the expression was "'somestring'")
- that val would not be evaluated correctly by anyone, so the user must be warned beforehand.
- besides, 

The story is different in 17, were attrs and raw modifiers have been merged and have taken the form of python expressions

That doesn't make the second attempt elegant at all, I have other solutions (cleaning the context at write-time in studio).
What do you prefer @beledouxdenis ?

Reply:
The system uses a subset of python code that does not automatically involve a boolean. This change seems very complex to me for cases that can be resolved such as:
```xml
<group invisible="context.get('default_brol')"/>
```
becomes
```xml
<group invisible="bool(context.get('default_brol'))"/>
```
Furthermore, behavior could be altered in cases such as:
```xml
<label for="journal_id"
    groups="account.group_account_readonly"
    invisible="context.get('default_journal_id') and context.get('move_type', 'entry') != 'entry'"/>
```

Reply:
This is not intuitive and error prone IMO.
This is pseudo python code, and I expect it to work as such

Reply:
There is a **lot** of cases for `invisible="context.get(...`, and most of them don't resolve to `(1, 0, True, False, None)`

Expecting this to work like a python expression evaluation has always been the case, even before the `attrs` refactor